pkg/master/client_ca_hook.go

```diff
@@ -58,11 +58,17 @@ func (h ClientCARegistrationHook) PostStartHook(hookContext genericapiserver.Pos
 		return h.tryToWriteClientCAs(client)
 	})
 
-	// if we're never able to make it through initialization, kill the API server
-	if err != nil {
+	// if we got an error during initialization, kill the API server
+	if err != nil && err != wait.ErrWaitTimeout {
 		return fmt.Errorf("unable to initialize client CA configmap: %v", err)
 	}
 
+	// if we timed out trying to initialize, log the error and continue so the server
+	// will at least be healthy long enough for someone to possibly fix the underlying cause
+	if err == wait.ErrWaitTimeout {
```

Comment:
nits: a switch block could make it look cleaner

Reply:
I'm not really sure how to clean this up with a switch block. Every time I try it ends up looking more complicated, like this:

```
switch err {
case nil:
	return nil
case wait.ErrWaitTimeout:
	utilruntime.HandleError(...)
	return nil
default:
	return fmt.Errorf(...)
}
```

Did you have something in mind for how to do it?

Reply:
You could do:
```
switch err {
case wait.ErrWaitTimeout:
	utilruntime.HandleError(...)
	fallthrough
case nil:
	return nil
default:
	return fmt.Errorf(...)
}
```
I'm not saying it's better though :-)

Reply:
I've redone it with a switch statement